yolox/models/build.py

```diff
@@ -48,44 +52,59 @@ def create_yolox_model(
-    if pretrained and num_classes == 80:
-        weights_url = _CKPT_FULL_PATH[name]
-        ckpt = load_state_dict_from_url(weights_url, map_location="cpu")
-        if "model" in ckpt:
-            ckpt = ckpt["model"]
-        yolox_model.load_state_dict(ckpt)
+    assert name in _CKPT_FULL_PATH or name == "yolox_custom", \
+        f"user should use one of value in {_CKPT_FULL_PATH.keys()} or \"yolox_custom\""
+    if name in _CKPT_FULL_PATH:
+        exp: Exp = get_exp(exp_name=name)
+        exp.num_classes = num_classes
+        yolox_model = exp.get_model()
+        if pretrained and num_classes == 80:
+            weights_url = _CKPT_FULL_PATH[name]
+            ckpt = load_state_dict_from_url(weights_url, map_location="cpu")
+            if "model" in ckpt:
+                ckpt = ckpt["model"]
+            yolox_model.load_state_dict(ckpt)
+    else:
+        assert exp_path is not None, "for a \"yolox_custom\" model exp_path must be provided"
+        exp: Exp = get_exp(exp_file=exp_path)
+        yolox_model = exp.get_model()
+        if ckpt_path:
+            ckpt = torch.load(ckpt_path, map_location="cpu")
+            if "model" in ckpt:
+                ckpt = ckpt["model"]
+            yolox_model.load_state_dict(ckpt)
 
     yolox_model.to(device)
     return yolox_model
 
 
-def yolox_nano(pretrained=True, num_classes=80, device=None):
+def yolox_nano(pretrained: bool = True, num_classes: int = 80, device: str = None) -> nn.Module:
     return create_yolox_model("yolox-nano", pretrained, num_classes, device)
 
 
-def yolox_tiny(pretrained=True, num_classes=80, device=None):
+def yolox_tiny(pretrained: bool = True, num_classes: int = 80, device: str = None) -> nn.Module:
     return create_yolox_model("yolox-tiny", pretrained, num_classes, device)
 
 
-def yolox_s(pretrained=True, num_classes=80, device=None):
+def yolox_s(pretrained: bool = True, num_classes: int = 80, device: str = None) -> nn.Module:
     return create_yolox_model("yolox-s", pretrained, num_classes, device)
 
 
-def yolox_m(pretrained=True, num_classes=80, device=None):
+def yolox_m(pretrained: bool = True, num_classes: int = 80, device: str = None) -> nn.Module:
     return create_yolox_model("yolox-m", pretrained, num_classes, device)
 
 
-def yolox_l(pretrained=True, num_classes=80, device=None):
+def yolox_l(pretrained: bool = True, num_classes: int = 80, device: str = None) -> nn.Module:
     return create_yolox_model("yolox-l", pretrained, num_classes, device)
 
 
-def yolox_x(pretrained=True, num_classes=80, device=None):
+def yolox_x(pretrained: bool = True, num_classes: int = 80, device: str = None) -> nn.Module:
     return create_yolox_model("yolox-x", pretrained, num_classes, device)
 
 
-def yolov3(pretrained=True, num_classes=80, device=None):
+def yolov3(pretrained: bool = True, num_classes: int = 80, device: str = None) -> nn.Module:
     return create_yolox_model("yolox-tiny", pretrained, num_classes, device)
```

Comment:
Here is a bug, here, plz also help to fix this. The string here should be `yolov3`

Reply:
Fixed this one, but I came from the [original code](https://github.com/Megvii-BaseDetection/YOLOX/blob/c4298f800d99ddc930267d8503d49b4dc06cff48/yolox/models/build.py#L91). Wonder if it should be fixed with a separate pr.